concent_api/core/tasks.py

```diff
@@ -34,7 +35,8 @@
 @transaction.atomic(using='control')
 def upload_finished(subtask_id: str):
     try:
-        subtask = Subtask.objects.get(subtask_id=subtask_id)
+        with database_lock_rows(Subtask, subtask_id) as database_lock:
```

Comment:
it is in one place now as a example, after fixing all suggestions will be done in all places where required